frontend/src/components/pages/CreateReview/CreateReviewPage.tsx

```diff
@@ -78,6 +115,62 @@ const CreateReview = (): React.ReactElement => {
     addBook(book);
   };
 
+  /**
+   * Function to be called when the user clicks the publish button.
+   */
+  const handlePublish = () => {
+    setReviewError(false);
+    setBylineError(false);
+
+    // Check if all fields are filled in
+    if (review !== "" && review !== "<p><br></p>" && byline !== "") {
+      setShowPublishModal(true);
+    } else {
+      if (review === "" || review === "<p><br></p>") {
+        setReviewError(true);
+      }
+      if (byline === "") {
+        setBylineError(true);
+      }
+    }
+  };
+
+  /**
+   * Function to be called when the review is published.
+   */
+  const onPublish = () => {
+    // check if all fields have been filled in
+    if (review !== "" || byline !== "" || books.length !== 0) {
+      // publish review
+      if (authenticatedUser?.id) {
```

Comment:
Do we need to check the `id` here? Is there ever a case where it'd be null/undefined?

Reply:
its so typescript knows authenticatedUser isn't null